### PR TITLE
fix: Bumped eslint, prettier-eslint to latest with updates

### PR DESCRIPTION
**Pull request by jmenglis**
Bumped eslint to 4.5.0 and prettier-eslint to 6.4.2. Updated eslint-config-kentcdodds to 12.4.2.
Added some changes to package.json to correct new formatting of error rules form eslint.  Imported
Config and Lint from eslint/lib to pass into the new ConfigFile.load(file, context)

fix: update to eslint 4.0 #94

**What**:

Bumped eslint to 4.5.0 and prettier-eslint to 6.4.2. Updated eslint-config-kentcdodds to 12.4.2. Added some changes to package.json to correct new formatting of error rules form eslint.  Imported Config and Lint from eslint/lib to pass into the new ConfigFile.load(file, context)

**Why**:

Want to get prettier-eslint-cli to the latest version of eslint and prettier-eslint

**How**:

Updated packages.  Added some additional logic to load method for loading config.  Updated eslint rules to include new eslint standards for semi and object curly braces.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -75,6 +75,19 @@
...
+          "max": 16
+        }
+      ],
+      "semi": [
+        "error",
+        "never"
+      ],
+      "object-curly-spacing": [
+        "error",
+        "never"
```
Comment (kentcdodds) [inline package.json]: Are these changes necessary?

File: src/format-files.js [hunk 2]
```diff
@@ -49,9 +51,10 @@ function formatFilesFromArgv({
     prettierLast,
     prettierOptions,
   }
-
+  const configContext = new Config({}, new Linter())
   if (eslintConfigPath) {
-    prettierESLintOptions.eslintConfig = ConfigFile.load(eslintConfigPath)
+    prettierESLintOptions.eslintConfig =
+      ConfigFile.load(eslintConfigPath, configContext)
```
Comment (kentcdodds) [inline src/format-files.js]: There's nothing that's changing about the external API right? So this is technically not a breaking change I think.

File: src/format-files.js [hunk 3]
```diff
@@ -49,9 +51,10 @@ function formatFilesFromArgv({
     prettierLast,
     prettierOptions,
   }
-
+  const configContext = new Config({}, new Linter())
```
Comment (kentcdodds) [inline src/format-files.js]: Also, it looks like `configContext` is only used within the `if` statement below, could we move this declaration to within that `if` statement?

File: package.json (hunk 1 above)
Comment (jmenglis) [reply package.json]: The tests do not pass without these changes.  I took a look at the [config repo](https://github.com/kentcdodds/eslint-config-kentcdodds) and did not see the cases for this.  Therefore when the `npm start validate` ran the files were handled according to the rules but two failed to occur.  The removal of a `;` and the spacing around an object `import { something } from 'something` vs `import {something} from 'something'`.  The test required tha latter to pass.
